opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -55,3 +60,48 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+
+
+class ResourceDetector:
+    def __init__(self, raise_on_error=False):
+        self.raise_on_error = raise_on_error
+
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        return _EMPTY_RESOURCE
+
+
+class OTELResourceDetector:
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        env_resources_items = os.environ.get("OTEL_RESOURCE")
+        env_resource_map = {}
+        if env_resources_items:
+            env_resource_map = {
+                key.strip(): value.strip()
+                for key, value in (
+                    item.split("=") for item in env_resources_items.split(",")
+                )
+            }
+        return Resource(env_resource_map)
+
+
+def get_aggregated_resources(
```

Comment:
I gave https://github.com/open-telemetry/oteps/pull/111 a full read, looks it needs
1. Timeout param
2. Can (optionally) run the individual detectors in parallel. This would make the timeout easy to implement I think :)

Reply:
Added. To be honest I'm not too familiar with multi thread stuff, would like to get some more eyes on it.